### ISSUE TITLE
Package assignment

### ISSUE THREAD

**Issue by theonejonahgold** [cmda-be/course-17-18#375]
# My package assignment
[Here](https://github.com/theonejonahgold/censor-swears) is my repo for the package assignment. I had a lot of fun doing the assignment, and learnt quite a lot from it!
## I used
- `xo` for code linting. I think the interface is really nice, and it has a lot of opinions I agree with.
- `jest` for unit testing. It's extremely easy to set-up and get started with.
## My feedback
I don't really have any negative feedback. The assignment was clear with plenty of direction **AND** room for tinkering. I really like these sort of assignments, so maybe you should take my 'feedback' with a grain of salt. 😅

**Comment by wooorm** [comment]
@theonejonahgold Awesome! This looks really great :) [Here’s how I did it](https://github.com/wooorm/f-ck), let me know if you see something weird and would like me to elaborate!

What did you think of the other packages used for inspiration? repeat-string, skin-tone, and my own longest-streak? Which did you like better?

It’s also cool that you used Jest! Maybe you’d like to try out using Travis in the future on a package, or this one even, to see what that gives you?

>  I had a lot of fun doing the assignment, and learnt quite a lot from it!

> I don't really have any negative feedback. The assignment was clear with plenty of direction AND room for tinkering. I really like these sort of assignments, so maybe you should take my 'feedback' with a grain of salt. 😅

Glad to hear!! 👍

Some more points

* I saw you’re using a `package-lock.json`. What do you think of those? I’m especially interested in whether `package-lock`s are needed for libraries in comparison to applications. (I personally don’t include them)
* I like that you linked to code from the API docs!
* For gitignores, the big standard Node one may work, but I personally like to include just what I need, to keep them a bit smaller
* [What’s this](https://github.com/theonejonahgold/censor-swears/blob/7b06fd55df7a399d63f5f648629af059650da4e4/.editorconfig#L5) in your `.editorconfig`? Why not one asterisk? 🤔

Anyway,  let me know if you find something interesting to do in the open source world and need an SRP! 😉

**Comment by theonejonahgold** [comment]
> I saw you’re using a `package-lock.json`. What do you think of those?

I think they're useful either way, as they prevent inconsistent package data between different machines working on the same project. Especially with NPM 5.6.0, which I think introduced consistent `package-lock` generation across different Operating Systems.

> For gitignores, the big standard Node one may work, but I personally like to include just what I need, to keep them a bit smaller

I'll see what I can exclude, I normally just set and forget these unless I know I'm going to include VSCode configs on my local machine.

> What’s this in your .editorconfig? Why not one asterisk? 🤔

I used double asterisks because javascript can be found in subfolders. This way I know for sure subfolders are taken into account as well. 😄 

> Maybe you’d like to try out using Travis in the future on a package, or this one even, to see what that gives you?

I already implemented travis inside a [project I'm currently planning out](https://github.com/theonejonahgold/tablejs). I may contact you about that in the near future 😋
